cordova-lib/src/cordova/plugin.js

```diff
@@ -283,6 +283,31 @@ module.exports = function plugin(command, targets, opts) {
     }
 };
 
+/**
+ * Calls `platformApi.prepare` for each platform in project
+ *
+ * @param   {string[]}  platformList  List of platforms, added to current project
+ * @param   {string}    projectRoot   Project root directory
+ *
+ * @return  {Promise}
+ */
+function preparePlatforms (platformList, projectRoot) {
+    return Q.all(platformList.map(function(platform) {
+        // TODO: this need to be replaced by real projectInfo
+        // instance for current project.
+        var project = {
+            root: projectRoot,
+            projectConfig: new ConfigParser(cordova_util.projectConfig(projectRoot)),
+            locations: {
+                plugins: path.join(projectRoot, 'plugins'),
+                www: cordova_util.projectWww(projectRoot)
+            }
+        };
+
+        return platforms.getPlatformApi(platform).prepare(project);
+    }));
+}
+
```

Comment:
Logic looks good to me. Just one suggestion - the guts of `preparePlatforms()` is an exact duplicate of code within `prepare.js`, correct? Why not add a `preparePlatform()` method in `prepare.js`, and call that from within the loop here and within the loop in `prepare`?

I guess one argument against that would be that this logic will become pretty simplistic once there is a real `projectInfo` instance for the project, but on the other hand, sharing this code means you don't have to update that in two places. So... I guess up to you :smile:.